src/contracts/Repository/Values/Content/Query/Criterion/Image/FileSize.php

```diff
@@ -15,18 +15,28 @@ public function __construct(
         $minValue = 0,
         $maxValue = null
     ) {
-        if ($minValue > 0) {
-            $minValue *= 1024 * 1024;
-        }
-
         if ($maxValue > 0) {
-            $maxValue *= 1024 * 1024;
+            $maxValue = $this->convertToBytes($maxValue);
         }
 
         parent::__construct(
             $fieldDefIdentifier,
-            $minValue,
+            $this->convertToBytes($minValue),
             $maxValue
         );
     }
+
+    /**
+     * @param numeric $value
+     */
+    private function convertToBytes($value): int
+    {
+        $value *= 1024 * 1024;
+
+        if (is_float($value)) {
+            $value = (int)$value;
+        }
+
+        return $value;
```

Comment:
Shouldn't
```suggestion
        return (int)$value;
```
be enough in all cases?

Reply:
Yeah, I was asking for a `is_float` check when `null` value was still possible in certain places.